src/components/framework/src/controller/value_resolvers/request_body.cr

```diff
@@ -120,14 +128,54 @@ struct Athena::Framework::Controller::ValueResolvers::RequestBody
     object
   end
 
-  private def map_request_body(body : IO, klass : ASR::Serializable.class)
+  private def map_query_string(request : ATH::Request, parameter : ATH::Controller::ParameterMetadata)
+    return unless query = request.query
+    return if query.nil? && (parameter.nilable? || parameter.has_default?)
+
+    self.deserialize_form query, parameter.type
+  rescue ex : URI::SerializableError
+    raise ATH::Exception::BadRequest.new "Malformed query string.", cause: ex
+  end
+
+  private def map_request_body(request : ATH::Request, parameter : ATH::Controller::ParameterMetadata)
+    if !(body = request.body) || body.peek.try &.empty?
+      raise ATH::Exception::BadRequest.new "Request does not have a body."
+    end
+
+    # We have to use separate deserialization methods with the case such that a type that includes multiple modules is handled as expected.
+    case request.content_type_format
+    when "form"
+      self.deserialize_form body, parameter.type
+    when "json"
+      self.deserialize_json body, parameter.type
+    else
+      raise ATH::Exception::UnsupportedMediaType.new "Unsupported format."
+    end
+  rescue ex : JSON::ParseException | ASR::Exception::DeserializationException
+    raise ATH::Exception::BadRequest.new "Malformed JSON payload.", cause: ex
+  rescue ex : URI::SerializableError
+    raise ATH::Exception::BadRequest.new "Malformed www form data payload.", cause: ex
```

Comment:
This and the other are technically wrong as this error is only raised when the underlying data is missing a required property. But I'm going to leave it and just address this more holistically as part of https://github.com/athena-framework/athena/issues/290.